src/main/java/org/sagebionetworks/bridge/hibernate/HibernateTagDao.java

```diff
@@ -0,0 +1,75 @@
+package org.sagebionetworks.bridge.hibernate;
+
+import static org.sagebionetworks.bridge.BridgeConstants.API_MAXIMUM_PAGE_SIZE;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Resource;
+
+import org.springframework.stereotype.Component;
+
+import org.sagebionetworks.bridge.dao.TagDao;
+import org.sagebionetworks.bridge.models.Tag;
+
+@Component
+public class HibernateTagDao implements TagDao {
+
+    private HibernateHelper hibernateHelper;
+    
+    @Resource(name = "basicHibernateHelper")
+    final void setHibernateHelper(HibernateHelper hibernateHelper) {
+        this.hibernateHelper = hibernateHelper;
+    }
+    
+    int getPageSize() {
+        return API_MAXIMUM_PAGE_SIZE;
+    }
+    
+    @Override
+    public Map<String, List<String>> getTags() {
+        Map<String, List<String>> byGroup = new HashMap<>();
+        List<Tag> page = null;
+        int offset = 0;
+        do {
+            page = hibernateHelper.queryGet("from Tag", null, offset, getPageSize(), Tag.class);
```

Comment:
We'd be better off asking the database for all the tags instead of calling a paginated API in a loop.

Reply:
Removed paging